src/read.c

```diff
@@ -780,6 +782,20 @@ typedef struct avifDecoderData
     avifTileInfo color;
     avifTileInfo alpha;
     avifDecoderSource source;
+    // When decoding AVIF images with grid cells, use a single decoder instance for all the cells instead of creating a decoder
+    // instance for each cell. If that is the case, |codec| will be used by all the tiles.
+    //
+    // There are some edge cases where we will still need multiple decoder instances:
+    // * For animated AVIF, we will need two instances (one for the color planes and one for the alpha plane since they are both
+    //   encoded as separate video sequences). In this case, |codec| will be used for the color planes and |codecAlpha| will be
+    //   used for the alpha plane.
+    // * For grid images with multiple layers. In this case, each grid will need its own decoder instance since there would be
+    //   multiple layers in each grid. In this case, |codec| and |codecAlpha| are not used and each tile will have it's own
```

Comment:
1 & 3 Done.

2) I believe used is the right word, since in that case `codec` and `codecAlpha` are neither created nor used. Only `tile->codec` is used.